--- a/gevent/pywsgi.py	2017-04-14 11:12:45.926693294 +0800
+++ b/gevent/pywsgi.py	2017-04-14 11:12:39.270693022 +0800
@@ -411,9 +411,12 @@
             delta)
 
     def process_result(self):
+        tmp_str = ''
         for data in self.result:
             if data:
-                self.write(data)
+                tmp_str+=data
+        if len(tmp_str) != 0:
+            self.write(tmp_str)
         if self.status and not self.headers_sent:
             self.write('')
         if self.response_use_chunked:
